util/time-utils/src/lib.rs

```diff
@@ -0,0 +1,62 @@
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+/// Temporary trait for `SystemTime::checked_add` until `Rust 1.33` is available
+pub trait CheckedSystemTime {
+	/// Returns `Some<SystemTime>` when the result less or equal to `i32::max_value` to prevent `SystemTime` to panic because
+	/// it is platform specific, possible representations are i32, i64, u64 and Duration. `None` otherwise
+	fn checked_add(self, _d: Duration) -> Option<SystemTime>;
+	/// Returns `Some<SystemTime>` when the result is successful and `None` when it is not
+	fn checked_sub(self, _d: Duration) -> Option<SystemTime>;
+}
+
+impl CheckedSystemTime for SystemTime {
+	fn checked_add(self, dur: Duration) -> Option<SystemTime> {
+		let this_dur = self.duration_since(UNIX_EPOCH).ok()?;
+		let total_time = this_dur.checked_add(dur)?;
+
+		if total_time.as_secs() <= i32::max_value() as u64 {
```

Comment:
so basically we can only add a maximum of 68 years with this method, just curious why this limit.

Reply:
Right, good questions Seun :P 

It is because SystemTime represents seconds as `i32, i64, u64 or Duration` (traditional OS's represent those as time_t i32 or i64).

This basically prevents the overflow in easy-way (but limits all platforms with limit of 2038) i.e, not have to rely on any complicated conditional compilation.

But this should be removed/deprecated when `Rust 1.33` is available (2019-02-28)